contrib/pg_tde/src/access/pg_tde_xlog_encrypt.c

```diff
@@ -179,13 +179,14 @@ TDEXLogSmgrInit(void)
 								  ((key->rel_type & TDE_KEY_TYPE_WAL_ENCRYPTED && !EncryptXLog) ||
 								   (key->rel_type & TDE_KEY_TYPE_WAL_UNENCRYPTED && EncryptXLog))))
 	{
-		pg_tde_create_wal_key(
-							  &EncryptionKey, &GLOBAL_SPACE_RLOCATOR(XLOG_TDE_OID),
+		pg_tde_create_wal_key(&EncryptionKey, &GLOBAL_SPACE_RLOCATOR(XLOG_TDE_OID),
 							  (EncryptXLog ? TDE_KEY_TYPE_WAL_ENCRYPTED : TDE_KEY_TYPE_WAL_UNENCRYPTED));
+		EncryptionCryptCtx = NULL;
 	}
 	else if (key)
 	{
 		EncryptionKey = *key;
+		EncryptionCryptCtx = NULL;
```

Comment:
I think it's redundant to set the ctx to NULL here and in L184. This function called only once on the server start and not gonna to be called for new backends. Moreover EncryptionCryptCtx is set to NULL in L71. It's actually a bit confusing - "why do we do this?"